core/node/eth_sender/src/eth_fees_oracle.rs

```diff
@@ -0,0 +1,148 @@
+                previous_sent_tx.base_fee_per_gas,
+                base_fee_per_gas,
+            )?;
+        }
+
+        let mut priority_fee_per_gas = self.gas_adjuster.get_priority_fee();
+
+        if let Some(previous_sent_tx) = previous_sent_tx {
+            // Increase `priority_fee_per_gas` by at least 20% to prevent "replacement transaction under-priced" error.
+            priority_fee_per_gas = max(
+                priority_fee_per_gas,
+                (previous_sent_tx.priority_fee_per_gas * 6) / 5 + 1,
+            );
+        }
+
+        // Extra check to prevent sending transaction will extremely high priority fee.
+        if priority_fee_per_gas > self.max_acceptable_priority_fee_in_gwei {
+            panic!(
+                "Extremely high value of priority_fee_per_gas is suggested: {}, while max acceptable is {}",
+                priority_fee_per_gas,
+                self.max_acceptable_priority_fee_in_gwei
+            );
+        }
+
+        Ok(EthFees {
+            base_fee_per_gas,
+            blob_base_fee_per_gas: None,
+            priority_fee_per_gas,
+        })
+    }
+
+    fn verify_base_fee_not_too_low_on_resend(
+        &self,
+        tx_id: u32,
+        previous_base_fee: u64,
+        base_fee_to_use: u64,
+    ) -> Result<(), EthSenderError> {
+        let next_block_minimal_base_fee = self.gas_adjuster.get_next_block_minimal_base_fee();
+        if base_fee_to_use <= min(next_block_minimal_base_fee, previous_base_fee) {
+            // If the base fee is lower than the previous used one
+            // or is lower than the minimal possible value for the next block, sending is skipped.
+            tracing::info!(
+                "Base fee too low for resend detected for tx {}, \
+                 suggested base_fee_per_gas {:?}, \
+                 previous_base_fee {:?}, \
+                 next_block_minimal_base_fee {:?}",
+                tx_id,
+                base_fee_to_use,
+                previous_base_fee,
+                next_block_minimal_base_fee
+            );
+            let err = ClientError::Custom("base_fee_per_gas is too low".into());
+            let err = EnrichedClientError::new(err, "increase_priority_fee")
+                .with_arg("base_fee_to_use", &base_fee_to_use)
+                .with_arg("previous_base_fee", &previous_base_fee)
+                .with_arg("next_block_minimal_base_fee", &next_block_minimal_base_fee);
+            return Err(err.into());
+        }
+        Ok(())
+    }
+}
+
+impl EthFeesOracle for GasAdjusterFeesOracle {
+    fn calculate_fees(
+        &self,
+        previous_sent_tx: &Option<TxHistory>,
+        has_blob_sidecar: bool,
+        time_in_mempool: u32,
+    ) -> Result<EthFees, EthSenderError> {
+        match has_blob_sidecar {
+            true => self.calculate_fees_with_blob_sidecar(previous_sent_tx),
+            false => self.calculate_fees_no_blob_sidecar(previous_sent_tx, time_in_mempool),
+        }
```

Comment:
nit: IMO using `if` is more appropriate than matching boolean

Reply:
indeed, corrected